### DISCUSSION TITLE
Modify when ViewMode changes

### DISCUSSION THREAD

**Discussion by corbin-poteet** [Sixze/ALS-Refactored#56] [Ideas]
When you change ViewMode from Third to First person, the camera takes a bit of time to travel from the Third Person transform to the First Person transform, but the variable ViewMode changes immediately. If you want to do something like hide the head of a modular character on client side when a character is in first person, there's a tiny window where ViewMode is set to First Person, but the camera is still transitioning from Third to First Person, resulting in the user being able to see the character with no head.

It'd be nice to have the ViewMode reflect when the float FirstPersonOverride in the AlsCameraComponent. In other words, if ViewMode was only set to FirstPerson when FirstPersonOverride is >= to 1.0. This would help with modular characters and handling the hiding of meshes when in First Person.

I'm not sure how to go about this, possibly add a "Transition" state to EViewMode or just a bTransitioning boolean to indicate when the camera is changing. Or rather than changing ViewMode, an "OnViewModeChanged" function that's called whenever the camera starts transitioning from first to third person, and finishes transitioning from third to first that we can override in child classes of the character.

**Comment by Sixze** [comment]
You can always read current value of the `FirstPersonOverride` curve to decide when to hide the head mesh, see how this curve is used in the `UAlsCameraComponent::TickCamera()` function, this should be enough for your purposes.
